runtime/Go/antlr/utils.go

```diff
@@ -385,3 +400,36 @@ func TitleCase(str string) string {
 	//	})
 
 }
+
+// murmur hash
+const (
+	c1_32 = 0xCC9E2D51
+	c2_32 = 0x1B873593
+	n1_32 = 0xE6546B64
+)
+
+func initHash(seed int) int {
+	return seed
+}
+
+func update(h1 int, k1 int) int {
+	k1 *= c1_32
```

Comment:
So many unexplained things. Is there a murmur hash already implemented that we can use? Something like https://gowalker.org/github.com/spaolacci/murmur3? Otherwise, this will need a lot of comments to explain what's going on.